### ISSUE TITLE
Create demo + demo video

### ISSUE THREAD

**Issue by JeffersonAero** [RMDLO/COCOpen-OpenCV#15]
Note: The submission is the COCOpen-OpenCV github repository. The code submission does not include training detectron2. The code submission should include a `demos/visualize.py` file which visualizes a subset of the generated dataset so that users can see for themselves the annotations produced by COCOpen. This file will be based on the [`Dataset_Verification.ipynb`](https://colab.research.google.com/drive/19rgHLHLHD5l3FWaswqwK30RT4OZy1WJi?authuser=1#scrollTo=up_KfLORZn94), but it will generate the mask visualization images in a python script and save images to a folder (make sure to add the folder to the `.gitignore`!). Will also need to update the `environment.yml` file to install detectron2 dependencies.

The demo includes training detectron2, generating a model, and qualitatively (and quantitatively?) demonstrating the performance of the model. All code for this demo is not part of the COCOpen-OpenCV repository.

The demo video should be ~5 minutes in duration and must:
1. Explain the algorithm. The explanation will include how it works, the sequential flow of data (steps of the process), approximate run time of each step, and memory requirements (time and space complexity).
2. Demonstrate the impact of the algorithm. We could include a demonstration of wire segmentation from a point cloud in real time, and a reach goal could include some type of real-time perception+wire manipulation with the robot.

**Comment by JeffersonAero** [comment]
Dataset verification currently requires installation of detectron2 in the conda environment. Can we split out visualization functions so that users do not have to install detectron2 (which also requires installation of pytorch)?

**Comment by JeffersonAero** [comment]
Created conda environment installation .yml comprising detectron2, pytorch, torchvision, and torchaudio (cpu-only version) + all cocopen-env dependencies and committed to `demo_environment.yml`. The cocopen dependencies are still in `environment.yml` in `config/`.

Install detectron2:
`conda install -c conda-forge detectron2`
Install pytorch:
`conda install pytorch torchvision torchaudio cpuonly -c pytorch`

**Comment by JeffersonAero** [comment]
detectron2 will require its own [installation instructions](https://detectron2.readthedocs.io/en/latest/tutorials/install.html) for the demo. It will not install in the conda environment. It can be installed with:

`python -m pip install 'git+https://github.com/facebookresearch/detectron2.git'`

![image](https://user-images.githubusercontent.com/40707210/206581725-4d9e700b-9203-46ff-b2d0-7414cecc15ba.png)

**Comment by JeffersonAero** [comment]
Created [`demo/` folder](https://github.com/RMDLO/COCOpen-OpenCV/tree/review/demo) with visualization of annotations for the auto-generated images. Added visualization to repo README.md.

**Comment by JeffersonAero** [comment]
Training detectron2 will require installation of pytorch with cuda enabled in the conda environment. [Here](https://pytorch.org/get-started/previous-versions/) are the available previous versions of pytorch (which are [compatible with detectron2](https://detectron2.readthedocs.io/en/latest/tutorials/install.html)). I plan to create a separate conda environment for training detectron2 (separate environment from visualizing results produced by cocopen or from generating a dataset with cocopen).
